### ISSUE TITLE
[Patterns] What should the result be if two distributed sets are equal except values are on different locales?

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#18965]
This issue is part of a series of issues to design the interface for collections across serial, parallel, and distributed contexts. Our goal is to determine what we think is reasonable to support and what doesn't seem useful. Additional patterns welcome (but it would be best to put them in their own issue for discussion, likely)

```chapel
var s1 = new DistributedSet(...);
var s2 = new DistributedSet(...);

... // Intentionally leaving "adding elements on different locales" vague - might not be specified by the user, might just happen

writeln(s1 == s2); // Should return true regardless, and ideally be efficient.
```

**Comment by mppf** [comment]
> Should return true regardless, and ideally be efficient.

That makes sense to me.

**Comment by vasslitvinov** [comment]
> Should return true regardless, and ideally be efficient.

Agreed. This is similar to `==` on ranges, which indicates whether the two ranges define the same sequence of integers, regardless of the representation. Cf. `ident` / `===` requires identical representation of its arguments.
